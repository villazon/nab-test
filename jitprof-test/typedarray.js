var a = [1,2,3];
var b = [4];
b.push("a");

for(var i = 0; i < 1000; i++){
  a.push(i);
}
