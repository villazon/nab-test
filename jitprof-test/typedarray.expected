gathering data...
-------------Fix Array Refactor Report-------------
Array created at the following locations may be special-typed:
location: (jitprof-test/typedarray.js:1:71:1:78)
	[Oper-Count]:	2001
	[Max-Indices]:	["1003"]
	[Refactor-Opts]: ["uint16arr","uint32arr","int16arr","int32arr","float32arr","float64arr"]
	[Func-Used]: ["push"]
	[Prop-Get]: ["push"]
[****]typedArray: 1
---------------------------------------------------
Following arrays can not be typed:
[x]	(jitprof-test/typedarray.js:2:9:2:12)	array stores non-numeric elements
